### PR TITLE
layout: Implement inline margins.

### PR DESCRIPTION
**Pull request by pcwalton**
Improves the Google SERPs.

r? @glennw

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/5940)

### CODE REVIEW

**Comment by glennw** [comment]
@pcwalton: Ran the wpt tests on your branch, and got a failure in:

/html/rendering/replaced-elements/images/space.html

**Comment by SimonSapin** [comment]
<sup>**[components/style/properties.mako.rs, line 5719 \[r1\]](https://reviewable.io:443/reviews/servo/servo/5940#-JoXf_ZqG8hdF0KKwBbx)** ([raw file](https://github.com/servo/servo/blob/0b49ee4aedcd7c96252b81717b37fbba2331b350/components/style/properties.mako.rs#L5719)):</sup>
This should be in logical directions.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/5940)

**Comment by pcwalton** [comment]
Marked test as failing and addressed review comments. r? @glennw

File: components/style/properties.mako.rs [hunk 1]
```diff
@@ -15,12 +15,11 @@ use std::hash::{Hash, Hasher};
 use std::sync::Arc;
 
 use util::fnv::FnvHasher;
-use util::logical_geometry::{WritingMode, LogicalMargin};
+use util::logical_geometry::{ LogicalMargin, PhysicalSide, WritingMode};
```
Comment (glennw) [inline components/style/properties.mako.rs]: nit: extra space here

**Pull request merged**